backend/dataall/core/environment/services/managed_iam_policies.py

```diff
@@ -119,32 +145,45 @@ def create_all_policies(self, managed) -> bool:
                         role_name=self.role_name,
                         policy_arn=f'arn:aws:iam::{self.account}:policy/{policy_name}',
                     )
-        except Exception as e:
-            raise e
+            except Exception as e:
+                logger.error(f'Error while creating and attaching policies due to: {e}')
+                raise e
         return True
 
     def delete_all_policies(self) -> bool:
         """
         Manager that registers and calls all policies created by data.all modules and that
         need to be deleted for consumption roles and team roles
         """
-        try:
-            for Policy in self.initializedPolicies:
-                policy_name = Policy.generate_policy_name()
+        for policy_manager in self.initializedPolicies:
+            policy_name_list = policy_manager.get_managed_policies()
+
+            # Check if policy with old naming format exists
```

Comment:
Although in theory it is impossible that there is an old_policy co-existing with new_indexed_policies, I would prefer to not make this assumption and remove 162 if-conditional as it is not needed and we ensure that no matter what we delete the old_policy

Reply:
There are situations in which we have seen some shares have old managed policies but no new managed policies. For example, an environment was onboarded in v2.6 then in v2.7 this IAM splitting change was deployd but they didn't migrate to new policy. Now in this case, they have a residual old managed policy which is not deleted. 
I don't see any problems arising due to this. But I thought this should also be deleted. 
Do you think this check needs to be removed ?

Reply:
Ahhhhh, I understand now! This is actually preventing the deletion of the old policy if the upgrade to indexed policies was not successful or did not happen for any reason. Seeing this point of view I think it is more important to ensure the sharing policy is not deleted than to clean it up. Let's leave the if-clause. This issue can be resolved